File: frontend/src/lib/types/storage.ts
export type StorageStatus = {
  is_configured: boolean;
  vault_path: string | null;
  library_path: string;
  database_url: string;
  vector_store_path: string | null;
  vector_store_provider: string | null;
  validation_message: string | null;
};

export type VaultValidation = {
  is_valid: boolean;
  vault_path: string;
  message: string;
  error_code: string | null;
  received_path: string | null;
  normalized_path: string | null;
  validated_by: string;
  failed_check: string | null;
  system_error: string | null;
  is_docker_path_issue: boolean;
};

File: backend/schemas/storage.py
from pathlib import Path

from pydantic import BaseModel


class VaultValidationRequest(BaseModel):
    vault_path: Path


class VaultValidationResponse(BaseModel):
    is_valid: bool
    vault_path: Path
    message: str
    error_code: str | None = None
    received_path: str | None = None
    normalized_path: Path | None = None
    validated_by: str = "backend"
    failed_check: str | None = None
    system_error: str | None = None
    is_docker_path_issue: bool = False


class WorkspaceStorageUpdate(BaseModel):
    vault_path: Path


class StorageStatusResponse(BaseModel):
    is_configured: bool
    vault_path: Path | None
    library_path: Path
    database_url: str
    vector_store_path: Path | None
    vector_store_provider: str | None
    validation_message: str | None = None
